Core/COEditingContext.m

```diff
@@ -412,8 +416,23 @@ - (void)updateCrossPersistentRootReferencesToPersistentRoot: (COPersistentRoot *
 {
 	NSParameterAssert(aPersistentRoot != nil);
 	// NOTE: -delete/undeleteBranch: enforce !aBranch.isCurrentBranch already.
-	NSParameterAssert(aBranch == nil
-		|| (!aBranch.isCurrentBranch && aBranch.persistentRoot == aPersistentRoot));
+	// NOTE: This is used by lazy loading now, and the following assertion no longer holds.
+	// Is this a problem?
+	//NSParameterAssert(aBranch == nil
+	//	|| (!aBranch.isCurrentBranch && aBranch.persistentRoot == aPersistentRoot));
+	
+	// See documentation above
+	if (isDeletion)
+	{
+		// TODO: For an uncommitted persistent root and branch, could be better if
```

Comment:
I figured out why the tests were passing, even when isDeletion was incorrect. I found a HACK in -replaceReferencesToObjectIdenticalTo:withObject: which can now be removed without breaking the test suite.

```
		// HACK: Now using -replaceReferencesToObjectIdenticalTo in the lazy loadying system
		// rather than just deletion/undeletion, so we need the logic in -crossPersistentRootReferenceWithPath
		// that checks whether branches/persistent roots are deleted
		COObject *testReplacement = [self.editingContext crossPersistentRootReferenceWithPath: object shouldLoad: NO];
		if (testReplacement == nil)
		{
			return;
		}
```